pkg/controller/attach_detach.go

```diff
@@ -110,17 +111,52 @@ func (r *ReconcileAttachDetach) Reconcile(ctx context.Context, request reconcile
 
 	// deletion request
 	if objectDeletionRequested(azVolumeAttachment) {
+		r.SharedState.nodeDiskAvailabilityMapLock.Lock()
+		defer r.SharedState.nodeDiskAvailabilityMapLock.Unlock()
+		// aliceyu? delete azVolumeAttachment +1
+		remainingCapacity, nodeExists := r.SharedState.nodeDiskAvailabilityMap.Load(azVolumeAttachment.Spec.NodeName)
+		if nodeExists {
+			r.logger.V(2).Info("aliceyu? delete: node: " + azVolumeAttachment.Spec.NodeName + " count: " + strconv.Itoa(remainingCapacity.(int)) + " vol: " + azVolumeAttachment.Spec.VolumeName + " role: " + string(azVolumeAttachment.Spec.RequestedRole))
+			r.SharedState.nodeDiskAvailabilityMap.Store(azVolumeAttachment.Spec.NodeName, remainingCapacity.(int)+1)
+		}
 		if err := r.removeFinalizer(ctx, azVolumeAttachment); err != nil {
+			// aliceyu? delete failed, -1
+			remainingCapacity, nodeExists := r.SharedState.nodeDiskAvailabilityMap.Load(azVolumeAttachment.Spec.NodeName)
+			if nodeExists {
+				r.logger.V(2).Info("aliceyu? delete: node: " + azVolumeAttachment.Spec.NodeName + " count: " + strconv.Itoa(remainingCapacity.(int)) + " vol: " + azVolumeAttachment.Spec.VolumeName + " role: " + string(azVolumeAttachment.Spec.RequestedRole))
+				r.SharedState.nodeDiskAvailabilityMap.Store(azVolumeAttachment.Spec.NodeName, remainingCapacity.(int)-1)
+			}
```

Comment:
could we update the remaining capacity of the node only after removing finalizer successfully?

Reply:
I couldn't see much difference between my current logic and yours. Do you have some specific reasons for that?

Reply:
Oh yeah, it should make no difference. Just to save one step to minus 1 back from the capacity if removing finalizer fails.

Reply:
I agree with @hccheng72 - it would simplify the code.

Reply:
Done